app/presenters/openid_connect_user_info_presenter.rb

```diff
@@ -25,7 +25,11 @@ def user_info
     info[:verified_at] = verified_at if scoper.verified_at_requested?
     if identity.vtr.nil?
       info[:ial] = if identity_proofing_requested_for_verified_user?
-                     Saml::Idp::Constants::IAL2_AUTHN_CONTEXT_CLASSREF
+                     if !resolved_authn_context_result.ialmax?
+                       resolved_authn_context_result.identity_proofing_names.first ||  Saml::Idp::Constants::IAL2_AUTHN_CONTEXT_CLASSREF
+                     else
+                      Saml::Idp::Constants::IAL2_AUTHN_CONTEXT_CLASSREF
+                     end
```

Comment:
I have a sneaking suspicion that this will break _something_, but I'm not sure what.

Thoughts, @jmhooper or @Sgtpluck?

Reply:
I find this implementation a little tough to follow, but I'll remark that it should return the "highest" value that was performed in the transaction. This is similar to the logic in `selected_vtr_parser_result_from_vtr_list`. E.g.:

- If biometric was requested and performed, then we return a value to indicate we performed biometric comparison.
- If identity proofing was requested and performed then we return a value to indicate we performed identity proofing
- If we did not perform identity proofing it is IAL1

Reply:
it was a little tough to write tests for this, so i decided to move the logic to the AuthnContextResolver, let me know what you think!